pkg/operator/operator.go

```diff
@@ -194,12 +103,10 @@ type Reporting struct {
 	importers   map[string]*prestostore.PrometheusImporter
 }
 
-func New(logger log.FieldLogger, cfg Config) (*Reporting, error) {
-	if err := cfg.APITLSConfig.Valid(); err != nil {
-		return nil, err
-	}
-	if err := cfg.MetricsTLSConfig.Valid(); err != nil {
-		return nil, err
+func New(logger log.FieldLogger, cfg Config) (ReportingOperator, error) {
```

Comment:
This is unrelated to the changes you've made so far to this constructor, but I would generally expect to see the initialization of all of the clientsets we may need throughout the operator, in something like cmd/reporting-operator/start.go instead of the actual constructor declaration. WDYT?

Reply:
> WDYT

Yes, after reducing the file I think there are a few more follow ons we can do.  (Side goal is to also keep these refactor commits small and reviewable so some may be better in follow on PRs)

I feel like a lot that happens in the private method could be moved into `New` or refactored into helpers that `New` just orchestrates for better readability so +1 to that suggestion

Reply:
don't think I read this right the first time.  I would need to look a bit more at that suggestion.  I think I agree - let the operator take the client as configuration and let the caller determine how they want to create it.  All we really care about in the context of the operator is the interface methods, not how the client was instantiated.

Reply:
It looks like the MCO does something similar but abstracts it a bit more using a custom type: https://github.com/openshift/machine-config-operator/blob/master/cmd/machine-config-controller/start.go